src/alloc/allocext/linux.rs

```diff
@@ -1,18 +1,25 @@
 extern crate std;
+use libc::{MAP_ANONYMOUS, MAP_FIXED, MAP_NORESERVE, MAP_SHARED, PROT_NONE};
+
 use self::std::process::abort;
 use crate::{alloc::*, Prot};
 use core::mem::{self, size_of};
-use core::ptr::{self, NonNull};
+use core::ptr::{self, null_mut, NonNull};
 use core::slice;
 
 use self::memfd_secret_alloc::*;
 
 mod memfd_secret_alloc {
+    use libc::{MAP_LOCKED, MAP_POPULATE};
+
     use super::*;
     use core::convert::TryInto;
 
     #[inline]
-    pub unsafe fn alloc_memfd_secret(size: usize) -> Option<(NonNull<u8>, libc::c_int)> {
+    pub unsafe fn alloc_memfd_secret(
+        size: usize,
+        ptr: Option<*mut libc::c_void>,
```

Comment:
It seems that no call is None?

Reply:
This is an internal function and no user will call it directly. we don't have to keep Option.